### PR TITLE
Skip rows with extra commas

### PR DESCRIPTION
**Pull request by amichal**
- [x] Add logic to skip over CSV lines that are too long/short
- [x] Log them to `LoadError`
- [x] Adjust `total_lines` and `lines_loaded` accordingly. `total_lines` now counts all read CSV lines **including the header**, `lines_loaded` counts only data rows we did not skip
- [x] Tests of the above
- [x] Adjust log and notifier output with a better summary table which includes wall time, records per second and %CPU utilization for various steps
- [x] get more specific on our `rescue`s when trying to recover from a bad batch/line/record in `import!`

### CODE REVIEW

**Comment by amichal** [comment]
@eanders I **did not** change the behaviour of padding out lines that are too short. Doing so would break this existing test

```
Failures:

  1) Validate import files includes expected assessments validations
     Failure/Error: expect(HmisCsvValidation::InclusionInSet.where(source_type: 'HmisCsvTwentyTwenty::Loader::Assessment').count).to eq(6)
     
       expected: 6
            got: 3
     
       (compared using ==)
     # ./drivers/hmis_csv_twenty_twenty/spec/models/validations_spec.rb:61:in `block (2 levels) in <main>'
```

File: drivers/hmis_csv_twenty_twenty/app/models/hmis_csv_twenty_twenty/loader/loader.rb [hunk 1]
```diff
@@ -206,29 +206,55 @@ def load_source_file_pg(read_from:, klass:)
...
-            # SLOW_CHECK;  klass.create! data
+          begin
+            parser = CSV.new(read_from, headers: false, liberal_parsing: true)
+            parser.each do |row|
+              # There were excess columns, probably due to an unquoted comma
+              if row.size > expect_col_count
+                row_errors << {
+                  file_name: base_name,
+                  message: 'Extra columns found',
+                  details: "lineno: #{parser.lineno}",
```
Comment (eanders) [inline drivers/hmis_csv_twenty_twenty/app/models/hmis_csv_twenty_twenty/loader/loader.rb]: Any reason we can't put "Line number: ..." here?

**Comment by eanders** [comment]
@amichal , it's probably fine to leave the pad-out.  It refuses to import the whole file if the header doesn't match right?  So this would only affect files where the header is correct, but one or more rows was missing a column, which seems unlikely.

File: drivers/hmis_csv_twenty_twenty/app/models/hmis_csv_twenty_twenty/loader/loader.rb (hunk 1 above)
Comment (amichal) [reply drivers/hmis_csv_twenty_twenty/app/models/hmis_csv_twenty_twenty/loader/loader.rb]: Nope... will change

**Comment by amichal** [comment]
@eanders The failing test above is an existing test case i inherited that has a short last row. 

Amusingly github points this out here https://github.com/greenriver/hmis-warehouse/blob/skip-rows-with-extra-commas/drivers/hmis_csv_twenty_twenty/spec/fixtures/files/validation_files/source/Assessment.csv#L1-L0

I'll poke around and see if I can change the test to count the row as invalid right now its counting 3 columns as containing bogus data

**Comment by amichal** [comment]
Ok so that was the only test case broken by excluding short lines.. pushing a new commit with that change. With it, these sorts of errors will now be caught in the loader and not the validator

**Comment by amichal** [comment]
Also ended up tweaking the summary formats to include more timing info sent out as notifications and *reducing* the types of errors we will recover from to only database update/save errors. We used to rescue `StandardError` which frequently masked coding errors. https://ruby-doc.org/core-2.5.1/Exception.html.... We might want to put some of those back like Range or divide by zero etc.

**Pull request merged**